app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntryEffectHandler.kt

```diff
@@ -110,4 +147,91 @@ object BloodPressureEntryEffectHandler {
       is Valid -> throw IllegalStateException("Date validation error cannot be $result")
     }.exhaustive()
   }
+
+  private fun createNewBpEntryTransformer(
+      userSession: UserSession,
+      facilityRepository: FacilityRepository,
+      patientRepository: PatientRepository,
+      bloodPressureRepository: BloodPressureRepository,
+      appointmentsRepository: AppointmentRepository,
+      userClock: UserClock,
+      scheduler: Scheduler
+  ): ObservableTransformer<CreateNewBpEntry, BloodPressureEntryEvent> {
+    return ObservableTransformer { createNewBpEntries ->
+      createNewBpEntries
+          .subscribeOn(scheduler)
```

Comment:
Subscribing to this on a different scheduler won't have any effect because we `flatMap` on a `Room` `Observable` query which switches the thread internally to Room's own scheduler. This is also seen in the `updateBpEntryTransformer` call since it doesn't accept a scheduler.

This is hidden behaviour, but since we are not unlikely to swap Room out for anything else, do you think we should remove the scheduler parameter from this method and just depend on Room to manage the threading for us at the moment?

Reply:
I could remove the `.subscribeOn` call.